torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -536,11 +537,25 @@ class TORCH_CUDA_API Allocate : public Expr {
     return buffer_->getDataType().value();
   }
 
+  Allocate* alias() const {
+    return alias_;
+  }
+
+  void setAlias(Allocate* alias) {
+    TORCH_INTERNAL_ASSERT(alias->getMemoryType() == memory_type_);
+    // TODO: TORCH_INTERNAL_ASSERT(alias->size() == size_);
+    alias_ = alias;
```

Comment:
Can there be multiple aliases? Should an alias only be allowed onto a non-aliased allocation?

Reply:
Yes. We could apply successive point-wise operations and reuse the same memory.